docs/functional_architecture.md

```diff
@@ -0,0 +1,92 @@
+# Functional Architecture
+
+## Enrich libzim functions
+
+zimscraperlib has primitives to enrich libzim functions with some operations which are known to be shared across scrapers. See `zim` module.
+
+## Handle videos
+
+zimscraperlib has primitives to manipulate videos with some operations which are known to be shared across scrapers. See `video` module.
+
+## Handle pictures
+
+zimscraperlib has primitives to manipulate pictures with some operations which are known to be shared across scrapers. See `image` module.
+
+## Store and rewrite mostly unmodified HTML, CSS and JS from online website
```

Comment:
Everything that follows is related to warc2zim-like context: storing generic web stuff into a ZIM. The language used is not adapted for a general, scraperlib-level document.
Instead of fixing it I suggest to put it all inside a section related to this goal which makes uses of rewrting.

Reply:
Yes, this is why I named this section like it is, and added the introduction speaking about rewriting. I don't get what more / different you would like to do.

Reply:
Do you mean I should rename the files? I've left them more generic because I think it could contain docs about other zimscraperlib functions as well at some point.

Reply:
All those points are on H2 level. I suggest to as an H2 and put the rest in H3.

As for the other two files, an introduction stating that currently this is only about the part in `rewritting` would remove confusion IMO

Reply:
I promise I already intended to move them to H3 ... but obviously it never went into practice and always stayed in my mind ^^